verifier_test.go

```diff
@@ -88,6 +111,12 @@ func TestNewVerifier(t *testing.T) {
 			alg:     0,
 			wantErr: true,
 		},
+		{
```

Comment:
Just a question on the overall method which is verified in this logic?

Should we not compare the exact error returned from the Verifier with the expected error which we are looking for.

It seems we only check whether we "wantErr" or not?

Reply:
It seems that we don't care about the detailed errors but has error or nil error.

Reply:
sure, that's how it is at the moment.  Yogesh has a point though that we could improve the precision of the checks.

Reply:
I will raise a separate issue to track this,

Reply:
Raised issue #118